byron/ledger/executable-spec/src/Ledger/Update.hs

```diff
@@ -1357,6 +1350,25 @@ instance STS UPIEND where
 instance Embed UPEND UPIEND where
   wrapFailed = UPENDFailure
 
+-- | Generate a protocol version endorsement for a given key, or 'Nothing' if no stable and
+-- confirmed protocol version update can be found.
+protocolVersionEndorsementGen
+  :: [(ProtVer, Set Core.VKeyGenesis)]
+  -- ^ Current set of endorsements
+  -> Gen (Maybe ProtVer)
+protocolVersionEndorsementGen endorsementsList =
+  if null mostEndorsedProposals
+  then pure Nothing
+  else Just <$> Gen.element mostEndorsedProposals
+  -- Take the top 5 most voted proposals, and endorse them. The constant 5 is determined arbitrarily
+  -- here.
+  where
+    mostEndorsedProposals :: [ProtVer]
+    mostEndorsedProposals = sortOn (second length) endorsementsList
```

Comment:
Instead of `sortOn` and `reverse`, you can use `Down` from `Data.Ord`, so
```
sortOn (Down . second length) endorsementsList & take 5 & fmap fst
```

Reply:
Nice.